kahypar/application/kahypar.cc

```diff
@@ -24,17 +24,28 @@
 #include "kahypar/io/hypergraph_io.h"
 #include "kahypar/partitioner_facade.h"
 
+#include "kahypar/utils/signal_handling.h"
+
+#include <csignal>
+#include <functional>
+
+
+//not pretty. but signal handling needs it.
+kahypar::Context* kahypar::SerializeOnSignal::context_p = nullptr;
+kahypar::Hypergraph* kahypar::SerializeOnSignal::hypergraph_p = nullptr;
+
```

Comment:
That stuff works as expected.
As long as you don't do any serious work in the signal handler.
The serializer should be fine.
At least it has been in the experiments.
I've left this in, in addition to the refinement time limit, because there are instances where the time limit doesn't apply properly, e.g. still in IP or not checked frequently enough.